File: ruby/src/training_data_formatter.rb
# rbs_inline: enabled

require_relative './lib/format'

class TrainingDataFormatter
  include ::Lib::Format

  # @rbs path_to_csv_training_data: String
  # @rbs return: void
  def initialize(path_to_csv_training_data)
    @path_to_csv_training_data = path_to_csv_training_data
  end

  # @rbs path_to_json_training_data: String
  # @rbs return: void
  def export(path_to_json_training_data)
    File.write(filename(path_to_json_training_data), to_json(path_to_csv_training_data))
  end

  private

  attr_reader :path_to_csv_training_data

  # @rbs path_to_json_training_data: String
  # @rbs return: String
  def filename(path_to_json_training_data)
    File.join(path_to_json_training_data, "training_data_#{Time.now.strftime('%F%T').gsub(/[:\-]/, '')}.json")
  end
end

File: python/src/training_data_formatter.py
from format import __to_json__
import os
import datetime
import sys
sys.path.append('./src/lib')


class TrainingDataFormatter:
    def __init__(self, csv_training_data: str) -> None:
        self.csv_training_data = csv_training_data

    def export(self, dirname: str) -> None:
        with open(self.__filename__(dirname), 'w') as f:
            f.write(__to_json__(self.csv_training_data))

    # private

    def __filename__(self, dirname: str) -> str:
        return os.path.join(
            dirname, f'training_data_{
                datetime.datetime.now():%Y%m%d%H%M%S}.json')
